src/components/TimeBasedLineChart.js

```diff
@@ -0,0 +1,753 @@
+
+      return Math.ceil(value / multiplier) * multiplier;
+    });
+
+    minValue = minValue > 0 ? 0 : minValue;
+
+    const ticks = [maxValue, minValue];
+    const interval = (maxValue - minValue) / numTicks;
+
+    for (let current = minValue; current < maxValue; current += interval) {
+      ticks.push(current);
+    }
+
+    if (this.props.alwaysShowZeroYTick) {
+      if (ticks.indexOf(0) === -1) {
+        ticks.push(0);
+      }
+    }
+
+    return ticks;
+  }
+
+  _getXScaleFunction () {
+    let maxDate = this.props.data[this.props.data.length - 1].timeStamp;
+    let minDate = this.props.data[0].timeStamp;
+
+    maxDate = moment.unix(maxDate).endOf(this.props.rangeType).unix();
+    minDate = moment.unix(minDate).startOf(this.props.rangeType).unix();
+
+    return d3.time.scale()
+      .range([0, this.state.adjustedWidth])
+      .domain([minDate, maxDate]);
+  }
+
+  _getXScaleValue (value) {
+    const xScale = this._getXScaleFunction();
+
+    return xScale(value);
+  }
+
+  _getYScaleFunction () {
+    const maxValue = d3.max(this.props.data, d => {
+      const value = d.above ? d.value + d.above : d.value + 1000;
+      const multiplier = value < 0 ? -1000 : 1000;
+
+      return Math.ceil(value / multiplier) * multiplier;
+    });
+
+    let minValue = d3.min(this.props.data, d => {
+      const value = d.below ? d.value + d.below : d.value - 1000;
+      const multiplier = value < 0 ? -1000 : 1000;
+
+      return Math.ceil(value / multiplier) * multiplier;
+    });
+
+    minValue = minValue > 0 ? 0 : minValue;
+
+    return d3.scale.linear()
+      .range([this.state.adjustedHeight, 0])
+      .domain([minValue, maxValue]);
+  }
+
+  _getYScaleValue (value) {
+    const yScale = this._getYScaleFunction();
+
+    return yScale(value) - 10;
+  }
+
+  _renderChart () {
+    const data = this.props.data;
+
+    this._renderChartBase();
+
+    if (data.length > 0) {
+      this._renderLineChart();
+    }
+  }
+
+  _renderChartBase () {
+    const margin = this.props.margin;
+    const width = this.props.width;
+    const height = this.props.height;
+    const chart = d3.select(this.state.chartEl);
+    const data = this.props.data;
+
+    chart.style(styles.svg);
+
+    chart.selectAll('g').remove();
+
+    chart.attr('width', width)
+      .attr('height', height);
+
+    if (data.length > 0) {
+      chart.append('g')
+        .attr('class', 'x-axis')
+        .attr('transform', 'translate(' + margin.left + ',' + (height - margin.bottom) + ')');
+
+      chart.append('g')
+        .attr('class', 'y-axis')
+        .attr('transform', 'translate(' + (margin.left + 30) + ',' + (margin.top - 10) + ')');
+
+      chart.append('g')
+        .attr('class', 'grid-line')
+        .attr('transform', 'translate(' + (margin.left - 10) + ',' + (margin.top - 10) + ')');
+    } else {
+      chart.append('g')
+        .append('text')
+          .attr('transform', 'translate(' + ((width / 2) - 37) + ',' + (height / 2.5) + ')')
```

Comment:
inherited numbers from your line chart in insite/target line chart.  I can do some investigating and see what we can do.